base/setup/usetup/usetup.c

```diff
@@ -595,15 +595,18 @@ UpdateKBLayout(VOID)
 {
     PGENERIC_LIST_ENTRY ListEntry;
     LPCWSTR pszNewLayout;
+    NTSTATUS Status;
 
     pszNewLayout = MUIDefaultKeyboardLayout();
 
     if (LayoutList == NULL)
     {
-        LayoutList = CreateKeyboardLayoutList(SetupInf, DefaultKBLayout);
-        if (LayoutList == NULL)
+        Status = CreateKeyboardLayoutList(SetupInf, DefaultKBLayout);
```

Comment:
```
../../base/setup/usetup/usetup.c: In function 'UpdateKBLayout':
../../base/setup/usetup/usetup.c:604:16: error: assignment makes integer from pointer without a cast [-Werror]
```

Not all return values are NTSTATUSes.

https://git.reactos.org/?p=reactos.git;a=blob;f=base/setup/usetup/settings.h;hb=12654d28522a70618de162912f6a97c3e59c0ff1#l52

Reply:
`InfFile` is declared as *HINF* which is a handle to an INF file. Hm. Shall I use `HRESULT` as replacement for `NTSTATUS` instead?

Reply:
No. NTSTATUS values won't magically appear. You have to make those functions actually return NTSTATUS values (and change their return type).